### PR TITLE
Converted sharded test from py to go

### PR DESCRIPTION
**Pull request by arindamnayak**
Converted [sharded.py ](https://github.com/vitessio/vitess/blob/master/test/sharded.py) to GO.
Signed-off-by: Arindam Nayak <arindam.nayak@outlook.com>

### CODE REVIEW

File: go/test/endtoend/sharded/sharded_test.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+	exitcode, err := func() (int, error) {
+		clusterInstance = &cluster.LocalProcessCluster{Cell: cell, Hostname: hostname}
+		defer clusterInstance.Teardown()
+
+		// Start topo server
+		if err := clusterInstance.StartTopo(); err != nil {
+			return 1, err
+		}
+		_ = clusterInstance.VtctlProcess.ExecuteCommand("CreateKeyspace",
```
Comment (deepthi) [inline go/test/endtoend/sharded/sharded_test.go]: we should not be ignoring any returned errors in tests. We should return it from here.

File: go/test/endtoend/sharded/sharded_test.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+	rows, err = shard1Master.VttabletProcess.QueryTablet("select id, msg from vt_select_test order by id", keyspaceName, true)
+	assert.Nil(t, err)
+	assert.Equal(t, `[[INT64(1) VARCHAR("test 1")]]`, fmt.Sprintf("%v", rows.Rows))
+
+	rows, err = shard2Master.VttabletProcess.QueryTablet("select id, msg from vt_select_test order by id", keyspaceName, true)
+	assert.Nil(t, err)
+	assert.Equal(t, `[[INT64(10) VARCHAR("test 10")]]`, fmt.Sprintf("%v", rows.Rows))
+}
+
+func reloadAlias(t *testing.T, aliases ...string) {
```
Comment (deepthi) [inline go/test/endtoend/sharded/sharded_test.go]: This should be called reloadSchema (or reloadSchemas).

File: go/test/endtoend/sharded/sharded_test.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+		clusterInstance = &cluster.LocalProcessCluster{Cell: cell, Hostname: hostname}
+		defer clusterInstance.Teardown()
+
+		// Start topo server
+		if err := clusterInstance.StartTopo(); err != nil {
+			return 1, err
+		}
+		_ = clusterInstance.VtctlProcess.ExecuteCommand("CreateKeyspace",
+			"--sharding_column_name", "keyspace_id",
+			"--sharding_column_type", "uint64",
```
Comment (deepthi) [inline go/test/endtoend/sharded/sharded_test.go]: This is the old way, remove these two parameters.

File: go/test/endtoend/sharded/sharded_test.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+	_, err = shard2.Vttablets[1].VttabletProcess.QueryTablet(sqlSchemaReverse, keyspaceName, true)
+	assert.Nil(t, err)
+
+	reloadAlias(t,
+		shard1Master.Alias,
+		shard1.Vttablets[1].Alias,
+		shard2Master.Alias,
+		shard2.Vttablets[1].Alias)
+
+	_ = clusterInstance.VtctlclientProcess.InitShardMaster(keyspaceName, shard1.Name, cell, shard1Master.TabletUID)
```
Comment (deepthi) [inline go/test/endtoend/sharded/sharded_test.go]: Does this return an error? If yes, we should not ignore it.

**Review by deepthi — COMMENTED** [review]
We should rename this test. It seems to be testing various vtctl commands.

**Comment by arindamnayak** [comment]
> We should rename this test. It seems to be testing various vtctl commands.

I have implemented all PR comments.  

Regarding renaming the test: Mostly the python code uses `utils.run_vtctl` to run vtctlclient commands, whose go equivalent is `clusterInstance.VtctlclientProcess.ExecuteCommand` . The python version either returns some result or throws exception and if exception is expected, then only assert it else ignore. Where as in go function, it always return `error` object which is only set of error is present else it is nil. The difference in porting is that, in go version, I always do `assert.Nil(t, err)`  if no error is expected, whereas in python version after executing the vtctlclient command, it does not checks for no exception. If you think asserting error from vtctctlclient is not required, I can remove those.

**Review by deepthi — APPROVED** [review]
Looks good with the changes.
I suggest renaming the file to shared_keyspace_test.go and the test to TestShardedKeyspace.

**Comment by arindamnayak** [comment]
> Looks good with the changes.
> I suggest renaming the file to shared_keyspace_test.go and the test to TestShardedKeyspace.

I have renamed the file and function in the last commit. I am merging this.

File: go/test/endtoend/sharded/shared_keyspace_test.go [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+	assert.Nil(t, err)
+
+	rows, err := shard1Master.VttabletProcess.QueryTablet("select id, msg from vt_select_test order by id", keyspaceName, true)
+	assert.Nil(t, err)
+	assert.Equal(t, `[[INT64(1) VARCHAR("test 1")]]`, fmt.Sprintf("%v", rows.Rows))
+
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("ValidateSchemaShard", fmt.Sprintf("%s/%s", keyspaceName, shard1.Name))
+	assert.Nil(t, err)
+
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("ValidateSchemaShard", fmt.Sprintf("%s/%s", keyspaceName, shard1.Name))
```
Comment (ajeetj) [inline go/test/endtoend/sharded/shared_keyspace_test.go:153]: Typo? Should be `shard2.Name` instead of `shard1.Name`.

**Pull request merged**